test/preview/components/routers/test_conditional_router.py

```diff
@@ -0,0 +1,162 @@
+            {"condition": "{{streams|length < 2}}", "output_type": str},
+        ]
+        with pytest.raises(ValueError):
+            ConditionalRouter(routes)
+
+    @pytest.mark.unit
+    def test_mandatory_and_optional_fields(self):
+        """
+        Router accepts a list of routes with mandatory and optional fields
+        """
+        routes = [
+            {"condition": "{{streams|length < 2}}", "output": "query", "output_type": str, "output_slot": "test"},
+            {"condition": "{{streams|length < 2}}", "output": "query", "output_type": str},
+        ]
+
+        ConditionalRouter(routes)
+
+    @pytest.mark.unit
+    def test_mandatory_and_optional_fields_with_extra_fields(self):
+        """
+        Router accepts a list of routes with mandatory and optional fields but not if some new field is added
+        """
+
+        routes = [
+            {
+                "condition": "{{streams|length < 2}}",
+                "output": "query",
+                "output_type": str,
+                "output_slot": "test",
+                "bla": "bla",
+            },
+            {"condition": "{{streams|length < 2}}", "output": "query", "output_type": str},
+        ]
+
+        with pytest.raises(ValueError):
+            ConditionalRouter(routes)
+
+    @pytest.mark.unit
+    def test_router_initialized(self, routes):
+        router = ConditionalRouter(routes)
+
+        assert router.routes == routes
+        assert set(router.__canals_input__.keys()) == {"query", "streams"}
+        assert set(router.__canals_output__.keys()) == {"query", "streams"}
+
+    @pytest.mark.unit
+    def test_router_evaluate_condition_expressions(self, router):
+        # first route should be selected
+        kwargs = {"streams": [1, 2, 3], "query": "test"}
+        result = router.run(**kwargs)
+        assert result == {"streams": [1, 2, 3]}
+
+        # second route should be selected
+        kwargs = {"streams": [1], "query": "test"}
+        result = router.run(**kwargs)
+        assert result == {"query": "test"}
+
+    @pytest.mark.unit
+    def test_router_evaluate_condition_expressions_using_output_slot(self):
+        routes = [
+            {
+                "condition": "{{streams|length > 2}}",
+                "output": "streams",
+                "output_slot": "enough_streams",
+                "output_type": List[int],
+            },
+            {
+                "condition": "{{streams|length <= 2}}",
+                "output": "streams",
+                "output_slot": "insufficient_streams",
+                "output_type": List[int],
+            },
+        ]
+        router = ConditionalRouter(routes)
+        # enough_streams output slot will be selected with [1, 2, 3] list being outputted
+        kwargs = {"streams": [1, 2, 3], "query": "Haystack"}
+        result = router.run(**kwargs)
+        assert result == {"enough_streams": [1, 2, 3]}
```

Comment:
@ZanSara @masci This unit test above illustrates why we need an `output_slot` field in the route directive. Suppose we're using the streams variable as the sole basis for routing. How can we ensure that streams are routed to the correct branch in the router? We require an additional field within the route directive that allows us to define, if needed, a distinct output slot for every condition. This way, data can be routed to the correct branch of the DAG pipeline. Putting an accent on "if needed" here - that's why it is an optional field.